wgengine/magicsock/magicsock.go

```diff
@@ -562,6 +585,13 @@ func (c *Conn) setEndpoints(endpoints []string, reasons map[string]string) (chan
 		return false
 	}
 
+	c.lastEndpointsTime = time.Now()
+
+	for _, fn := range c.onEndpointRefreshed {
+		go fn()
```

Comment:
Mildly worried about no synchronization here (e.g. blocking on everyone finishing before returning). No specific reason why, fire-and-forget goroutines just make me nervous in general. YW to ignore that if you feel happy :)